fastapi/routing.py

```diff
@@ -404,18 +404,22 @@ def __init__(
         self.response_class = response_class
 
         assert callable(endpoint), "An endpoint must be a callable"
-        self.dependant = get_dependant(path=self.path_format, call=self.endpoint)
-        for depends in self.dependencies[::-1]:
-            self.dependant.dependencies.insert(
-                0,
-                get_parameterless_sub_dependant(depends=depends, path=self.path_format),
-            )
+        self.dependant = self.gen_dependant()
         self.body_field = get_body_field(dependant=self.dependant, name=self.unique_id)
         self.dependency_overrides_provider = dependency_overrides_provider
         self.callbacks = callbacks
         self.app = request_response(self.get_route_handler())
         self.openapi_extra = openapi_extra
 
+    def gen_dependant(self):
+        dependant = get_dependant(path=self.path_format, call=self.endpoint)
+        for depends in self.dependencies[::-1]:
+            dependant.dependencies.insert(
+                0,
```

Comment:
What is 0? Assign to a variable and pass the bound variable to the function. The variable name should explain what the 0 is.

Reply:
please look up to the deleted line 407-412, the function is just copy from the origin code and make it a function.

this function is generate dependant recursively.